lib/teleterm/clusters/cluster_kubes.go

```diff
@@ -111,22 +114,56 @@ type GetKubesResponse struct {
 }
 
 // reissueKubeCert issue new certificates for kube cluster and saves them to disk.
-func (c *Cluster) reissueKubeCert(ctx context.Context, kubeCluster string) error {
+func (c *Cluster) reissueKubeCert(ctx context.Context, kubeCluster string, mfaPrompt mfa.Prompt) (tls.Certificate, error) {
 	// Refresh the certs to account for clusterClient.SiteName pointing at a leaf cluster.
 	err := c.clusterClient.ReissueUserCerts(ctx, client.CertCacheKeep, client.ReissueParams{
 		RouteToCluster: c.clusterClient.SiteName,
 		AccessRequests: c.status.ActiveRequests.AccessRequests,
 	})
 	if err != nil {
-		return trace.Wrap(err)
+		return tls.Certificate{}, trace.Wrap(err)
 	}
 
-	// Fetch the certs for the kube cluster.
-	return trace.Wrap(c.clusterClient.ReissueUserCerts(ctx, client.CertCacheKeep, client.ReissueParams{
-		RouteToCluster:    c.clusterClient.SiteName,
-		KubernetesCluster: kubeCluster,
-		AccessRequests:    c.status.ActiveRequests.AccessRequests,
-	}))
+	proxyClient, err := c.clusterClient.ConnectToProxy(ctx)
+	if err != nil {
+		return tls.Certificate{}, trace.Wrap(err)
+	}
+	defer proxyClient.Close()
+
+	var mfaRequired bool
+	key, err := proxyClient.IssueUserCertsWithMFA(
+		ctx, client.ReissueParams{
+			RouteToCluster:    c.clusterClient.SiteName,
+			KubernetesCluster: kubeCluster,
+			RequesterName:     proto.UserCertsRequest_TSH_KUBE_LOCAL_PROXY,
+		},
+		mfaPrompt,
+		client.WithMFARequired(&mfaRequired),
+	)
+	if err != nil {
+		return tls.Certificate{}, trace.Wrap(err)
+	}
+
+	// TODO(ravicious): Add the equivalent of checkIfCertsAreAllowedToAccessCluster.
+
+	// TODO(ravicious): Share this logic with issueKubeCert in tool/tsh/common/kube_proxy.go.
+	if !mfaRequired {
+		if err := c.clusterClient.LocalAgent().AddKubeKey(key); err != nil {
+			return tls.Certificate{}, trace.Wrap(err)
+		}
+	}
+	cert, err := key.KubeTLSCert(kubeCluster)
+	if err != nil {
+		return tls.Certificate{}, trace.Wrap(err)
+	}
+	// Set leaf so we don't have to parse it on each request.
+	leaf, err := utils.TLSCertLeaf(cert)
+	if err != nil {
+		return tls.Certificate{}, trace.Wrap(err)
+	}
+	cert.Leaf = leaf
+
+	return cert, nil
```

Comment:
I'll tackle sharing code between tsh and teleterm within this PR, but I want to write some integration tests for that new lib/teleterm code first.